processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -524,6 +524,7 @@ public boolean isCacheable(GroupByQuery query, boolean willMergeRunners)
             .appendCacheables(query.getAggregatorSpecs())
             .appendCacheables(query.getDimensions())
             .appendCacheable(query.getVirtualColumns())
+            .appendCacheable(query.getLimitSpec())
```

Comment:
Does this also need to consider the `havingSpec` similar to the ResultLevelCacheKey below?

Reply:
havingSpec is used in post-processing step only, so it should not be required here